src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,51 @@
+export enum MetadataKind {
+  REFERRAL = 'referrer',
+}
+
+interface Metadata {
+  kind: MetadataKind
+  version: string
+}
+
+export interface ReferralMetadata extends Metadata {
+  referrer: string
+}
+
+export type MetadataDoc = {
+  [MetadataKind.REFERRAL]?: ReferralMetadata
+}
+
+export type AppDataDoc = {
+  version: string
+  appCode?: string
+  metadata: MetadataDoc
```

Comment:
I think this should be an array of Metadatas?, right?

Reply:
I don't know about the background behind changing the type from an array to an object. I can imagine better compile errors on repeated keys and also the impossibility of having repeated keys dynamically.

Reply:
As discussed with Anxo, this was a conscious decision made in the design doc for couple of reasons:
1. Enforces only one doc of a given type
2. Makes the doc more easily "indexable". It's easier to access a property by name rather than always looking it up in the array.